System/Posix/Semaphore.hsc

```diff
@@ -79,6 +80,20 @@ semWait (Semaphore fptr) = withForeignPtr fptr semWait'
     where semWait' sem = throwErrnoIfMinus1Retry_ "semWait" $
                          sem_wait sem
 
+-- | Lock the semaphore, blocking until it becomes available.
+--
+-- Unlike 'semWait', this wait operation can be interrupted with an
+-- EINTR signal.
```

Comment:
You mean a `SIGPIPE` signal? `eINTR` is the error code. But that's an implementation detail of the RTS, isn't it? Any `throwTo` can interrupt.

https://downloads.haskell.org/~ghc/9.2.5/docs/html/users_guide/exts/ffi.html#interruptible-foreign-calls

Reply:
> You mean a `SIGPIPE` signal? `eINTR` is the error code. But that's an implementation detail of the RTS, isn't it? Any `throwTo` can interrupt.
> 
> https://downloads.haskell.org/~ghc/9.2.5/docs/html/users_guide/exts/ffi.html#interruptible-foreign-calls

More precisely, actually interrupted by a `throwTo` from another thread. The use of `SIGPIPE` on Unix systems for this purpose is an implementation detail. And of course `EINTR` is not the signal but rather a feature of the return state.